### PR TITLE
refactor/47-utils

### LINKED ISSUE #47: getUserId.js isLoggedIn has a Cognitive Complexity of 25 (exceeds 5 allowed) and has 43 Lines of Code (exceeds 25 allowed)
[Code Climate Link](https://codeclimate.com/github/Lambda-School-Labs/nutrition-tracker-be-pt7/issues?category=complexity&engine_name%5B%5D=structure&engine_name%5B%5D=duplication)

Code Climate estimated time to fix: 3hours 

@Afrodo1 and I have brought discussed not really understanding what this code means with the use of the ctx parameter within this function.

This could be fixed with the resolution of Issue #43 as we are rehauling authentication

### PR DESCRIPTION
**Pull request by CodyBrewer**
# Description

Reduces the cognitive complexity and readibility of the utility functions, specifically the getUserId/isLoggedIn utility function.

Additional tests written for utility functions.

Fixes #47 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Change Status

- [x] Complete, tested, ready to review and merge
- [x] Complete, but not tested (may need new tests)
- [x] Incomplete/work-in-progress, PR is for discussion/feedback

## How Has This Been Tested?

- [x] previous unit tests in tests/Utilites.spec.js passing
- [x] Unit test written for formatDate.js function
- [x] Unit test written for generateToken.js function

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by CodyBrewer** [comment]
[Code Climate Score of D with 23 issues](https://trello.com/c/36uMpJRK/46-code-climate-score-of-d-with-23-issues)

**Review by CodyBrewer — COMMENTED** [review]
📖 self-review complete 📘

File: tests/Utilities.spec.js [hunk 1]
```diff
@@ -1,110 +1,148 @@
...
+        },
+        null,
+        null,
+        login
+        );
+        
+        expect(loggedIn).toBe('yay');
+    });
+  });  
+  describe('formatDate()', () => {
```
Comment (dave-vazquez) [inline tests/Utilities.spec.js:116]: Nice! Cleaned up testing output and wrote additional tests 👍

File: utils/authorization.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const jwt = require('jsonwebtoken');
+
+const authorization = (resolve, parent, args, ctx) => {
```
Comment (dave-vazquez) [inline utils/authorization.js:3]: 🎉

**Review by dave-vazquez — APPROVED** [review]

**Pull request merged**